src/chapters/thymeleaf-views/template-conditionals.rst

```diff
@@ -0,0 +1,237 @@
+Conditionals in a Template
+===========================
+
+In addition to iteration, Thymeleaf can also add or remove content on a
+webpage based on certain conditions. Going back to the coffee example, we could
+generate the ordered list ONLY IF ``coffeeOptions`` contains data. If the
+ArrayList is empty, then there is no need to include the ``<ol>`` element.
+Instead, the template could include a ``<p>`` element with text stating that
+there are no options to select.
+
+Just like the ``for/each`` syntax differs between Java and Thymeleaf, we need
+to examine how to include *conditionals* in our templates. The logic remains
+the same, but the implementation requires practice.
+
+Try It!
+--------
+
+Code along with the following video to gain more practice with Thymeleaf:
+
+   Video goes here...
+
+For your reference, here is an extension for some of the concepts presented
+in the clip. Note that the text is NOT intended as a replacement for the
+walkthrough. To get better at coding, you need to actually CODE instead of just
+reading about how to do it.
+
+Display Content ``if``
+-----------------------
+
+The general syntax for including a conditional in Thymeleaf is:
+
+.. sourcecode:: groovy
+
+   th:if = "${condition}"
+
+#. The ``th:if`` statement gets placed inside an HTML tag.
+#. ``condition`` represents a boolean variable provided by the controller.
+#. Alternatively, ``condition`` can be a statement that evaluates to ``true``
+   or ``false``.
+
+.. admonition:: Note
+
+   ``condition`` could also represent a function call that returns a boolean.
+   However, making this work properly is an advanced topic and is beyond the
+   scope of this course.
+
+If ``condition`` evaluates to ``true``, then Thymeleaf adds the HTML element to
+the webpage, and the content gets displayed in the view. If ``condition`` is
+``false``, then Thymeleaf does NOT generate the element, and the content stays
+off the page.
+
+.. admonition:: Example
+
+   Assume that ``coffeeOptions`` and ``userSelection`` represent an ArrayList
+   and String, respectively, that are passed in by the controller.
+
+   .. sourcecode:: html
+      :linenos:
+
+      <ol th:if = "${coffeeOptions.size() > 1}">
+         <li th:each="item : ${coffeeOptions}" th:text="${item}"></li>
```

Comment:
``th:each`` should be accompanied with a ``th:block``.

Reply:
``th:block`` has its own section later on the page.  I thought it might be better to discuss ``th:each`` first and then introduce the idea of the wrapper.

Reply:
I can see that! Since both ``th:block`` and ``th:each`` are introduced on the same page and we talk about the importance of using the two together, I can just see students wondering why ``th:block`` is not used here.